go/plugins/compat_oai/openai/openai.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openai
+
+import (
+	"context"
+
+	"github.com/firebase/genkit/go/ai"
+	"github.com/firebase/genkit/go/genkit"
+	"github.com/firebase/genkit/go/plugins/compat_oai"
+	openaiGo "github.com/openai/openai-go"
+	"github.com/openai/openai-go/option"
+)
+
+const provider = "openai"
+
+var (
+	supportedModels = map[string]ai.ModelInfo{
+		openaiGo.ChatModelGPT4oMini: {
+			Label:    "GPT-4o-mini",
+			Supports: compat_oai.Multimodal.Supports,
+		},
+	}
+
+	knownEmbedders = []string{
+		openaiGo.EmbeddingModelTextEmbedding3Small,
+		openaiGo.EmbeddingModelTextEmbedding3Large,
+		openaiGo.EmbeddingModelTextEmbeddingAda002,
+	}
+)
+
+type OpenAI struct {
+	Opts             []option.RequestOption
+	openAICompatible compat_oai.OpenAICompatible
+}
+
+// Name implements genkit.Plugin.
+func (o *OpenAI) Name() string {
+	return provider
+}
+
+func (o *OpenAI) Init(ctx context.Context, g *genkit.Genkit) error {
+	err := o.openAICompatible.Init(ctx, g)
```

Comment:
Nit: here and elsewhere, can do this which is more succinct and preferred:

```go
if err := o.openAICompatible.Init(ctx, g); err != nil {
     return err
}
```

Reply:
fixed here: https://github.com/kekoawong/genkit/pull/11